### PR TITLE
Fix PE-D bugs

### LINKED ISSUE #206: [PE-D][Tester E] App does not support "S/O" in names
**Steps to reproduce:** 
1. Add a user with the name "John Doe S/O John Deez"

It might affect users who want to add contact with "s/o" in their legal name

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: seanlim/ped#4

### LINKED ISSUE #214: [PE-D][Tester A] User Guide: Sort Command
**Description**

When reading the User Guide, when coming across the bullet point below, it may lead to the user believing that the command can just be run in one line to achieve that result, when in fact, the commands need to be run one after the other.

**Suggestion**
You could change the phrasing to something like 

> e.g. `find a/clementi` followed by `sort address`

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: OttoTan/ped#11

### LINKED ISSUE #217: [PE-D][Tester A] Sort Command (Parameters shown)
**Description**
Although the parameters provided are listed in the command summary, it is not listed in the feature list, as shown below

**Suggestions**
This may lead to the users being confused about what parameters are available for use, especially if they have not seen the command summary which is at the bottom of the guide. 

Perhaps you can also list out the parameters available at the beginning

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: OttoTan/ped#12

### LINKED ISSUE #227: [PE-D][Tester D] Add command does not let me add a name like Björn
Entered commands: add n/Björn Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 dob/1999-03-09 pri/HIGH income/2000 famsize/3 r/Promises to pay back next week. t/friends t/owesMoneyv,

add n/John Doe s/o p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 dob/1999-03-09 pri/HIGH income/2000 famsize/3 r/Promises to pay back next week. t/friends t/owesMoney

Maybe the UG should specify that names should be alphanumeric only? However I do feel that the app should not limit the users based on name.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: mingyang143/ped#2

### LINKED ISSUE #234: [PE-D][Tester C] Find command does not apply search for contacts on the main list
When using the Find command, it only searches within the currently filtered list rather than the full list of contacts. For instance, if I first use the Find command to search for a contact with the name "David," and then search for "Alice" afterward, the second search yields no results although "Alice" exists in the main contact list. This likely occurs because the Find command operates only on the currently-filtered list in view, not the full list of contacts.

This bug has been classified with a medium severity because it could affect the accuracy of overall contact searches and inconvenience users who want to do numerous consecutive searches overall for contacts, since they'd have to type 'list' before using Find every time. 

**Steps to reproduce:** 
1. Add contacts to the main list with names "David" and "Alice"

2. Use the Find command to search for "David." The contact list should display only "David."
`find n/David`

3. Without resetting the contact list, immediately use the Find command again to search for "Alice."
`find n/Alice`

**Expected output:** 
The contact list should display the contact(s) which contain the name "Alice"

**Actual output:**
"0 persons listed!" is shown in the command output box

<i>
  <video controls width="320" height="180">
    <source src="https://raw.githubusercontent.com/kikokidayo/ped/main/files/635616c6-6ead-4cf8-a00e-2dc3bd91a7f4.mov" type="video/mp4">
    Your browser does not support the video tag.
  </video>
  <br></i>

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: kikokidayo/ped#6

### PR DESCRIPTION
**Pull request by arjun2598**
* Fixes #206 
* Fixes #214 
* Fixes #217 
* Fixes #227 
* Fixes #234 

Notes:
1. Issue #234 is about find working only on the filtered list and not the entire list, hence it is tagged not in scope as the filtering could be enhanced in future to include both ways. However, it can be left out from planned enhancements if needed because I have added a warning related to this in the UG and the filtering works as intended by us. 
2. Added a screenshot under the Sort feature section. Can be changed to standardise with the other screenshots if needed.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -54,6 +54,54 @@ If done correctly, a GUI similar to the image below should appear in a few secon
...
+4. EMAIL: Emails should be of the format local-part@domain. **Most common emails will be accepted** but refer to the specifications below if you need more information: <br> Prefix: `e/`
+   * "local-part" can only contain alphanumeric characters and the following special characters: .+_- (Cannot start/end with any special characters)
+   * Domain name is made up of domain labels separated by periods. The domain name must: 
+     * End with a domain label at least 2 characters long
+     * Have each domain label start and end with alphanumeric characters
+     * Have each domain label consist of alphanumeric characters, and only separated by hyphens (if any)
+   * Valid Examples: `john.doe@example.com`, `jane-doe123@example.com`
+   * Invalid Examples: `john.doe.@example.com`, `john.doe@exa_mple.com`
+
+5. DATE OF BIRTH: Date of birth must not be a future date. Input must follow the format YYYY-MM-DD. <br> Prefix: `dob/`
```
Comment (Gra7ityIC3) [inline docs/UserGuide.md]: Could you change `YYYY-MM-DD` to `yyyy-MM-dd`?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -54,6 +54,54 @@ If done correctly, a GUI similar to the image below should appear in a few secon
...
+
+### Optional Fields:
+
+1. PRIORITY: Priority should be HIGH, MEDIUM, or LOW. Default: LOW <br> Prefix: `pri/` 
+
+2. INCOME: Income should be a non-negative decimal number. Default: 0 <br> Prefix: `income/`
+
+3. FAMILY SIZE: Family size should be a positive integer. Default: 1 <br> Prefix: `famsize/`
+
+4. TAGS: Tag names should be alphanumeric. Default: Empty <br> Prefix: `t/`
```
Comment (Gra7ityIC3) [inline docs/UserGuide.md]: I think the default value for tags can be removed.

**Review by Gra7ityIC3 — APPROVED** [review]
LGTM!

File: docs/UserGuide.md (hunk 1 above)
Comment (arjun2598) [reply docs/UserGuide.md]: Sure will make this change

File: docs/UserGuide.md (hunk 2 above)
Comment (arjun2598) [reply docs/UserGuide.md]: Will remove it

**Pull request merged**